extensions/mongoose/README.md

```diff
@@ -0,0 +1,160 @@
+# @loopback/mongoose
+
+## Stability: ⚠️Experimental⚠️
+
+> Experimental packages provide early access to advanced or experimental
+> functionality to get community feedback. Such modules are published to npm
+> using `0.x.y` versions. Their APIs and functionality may be subject to
+> breaking changes in future releases.
+
+[![LoopBack](<https://github.com/strongloop/loopback-next/raw/master/docs/site/imgs/branding/Powered-by-LoopBack-Badge-(blue)-@2x.png>)](http://loopback.io/)
+
+## ⚠️Warnings⚠️
+
+As of `0.0.1` of this extension, the type definitions of `@types/mongoose` are
+several signifigant minor versions out of date: it supports mongoose `5.7.12`,
+and the latest stable version of mongoose is `5.9.10`
+
+## Using your existing Mongoose Schemas in Loopback
+
+Using mongoose to devine your mongodb schemas is relatively straightforward,
+though at this early stage it's still a number of steps:
+
+- Import the component, and it's binding:
+  `import {LoopbackMongooseComponent, MongooseBindings} from '@loopback/mongoose'`
+
+- Register your configuration (see below)
+  ```ts
+  this.configure(MongooseBindings.COMPONENT).to(configuration);
+  ```
+- Add the component to your application
+  ```ts
+  this.component(LoopbackMongooseComponent);
+  ```
+- Finally, `@inject` your Entity Models for use
+  ```ts
+  public async MyService(@inject('loopback-mongoose-component.model.ModelName') ModelName) {
+    const result = await ModelName.findOne({});
+  }
+  ```
+
+## Configuration
+
+There are three pieces of information that the Mongoose Component needs in order
+to create a connection to your database and set up the mongoose models:
+
+- the `uri` of the connection
+- an array of Schemas
+- optionally, an array of Discriminator Schemas
+
+The configuration looks like the following:
+
+```ts
+ this.configure(MongooseBindings.COMPONENT).to({
+   uri: 'mongodb://localhost:27017',
+   connectionOptions: {} // anything valid to mongoose.createConnection(),
+   schemas: [
```

Comment:
Maybe it's better to declare `mongoose.schemas` as an extension point to register Mongoose schemas as extensions.